lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/ActiveContextManager.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using System.Globalization;
+using Org.Apache.REEF.Driver.Context;
+using Org.Apache.REEF.Driver.Evaluator;
+using Org.Apache.REEF.Utilities.Attributes;
+using Org.Apache.REEF.Utilities.Diagnostics;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver
+{
+    /// <summary>
+    /// Manages active contexts for the driver
+    /// </summary>
+    [NotThreadSafe]
+    internal sealed class ActiveContextManager
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(ActiveContextManager));
+        private readonly IDictionary<string, IActiveContext> _activeContexts = new Dictionary<string, IActiveContext>();
+        private readonly int _totalExpectedContexts;
+
+        /// <summary>
+        /// Constructor of ActiveContextManager
+        /// totalContexts specify the total number of expected active contexts that driver needs
+        /// </summary>
+        /// <param name="totalContexts"></param>
+        internal ActiveContextManager(int totalContexts)
+        {
+            _totalExpectedContexts = totalContexts;
+        }
+
+        /// <summary>
+        /// Returns the collection of IActiveContext
+        /// </summary>
+        internal ICollection<IActiveContext> ActiveContexts
```

Comment:
what if someone does the following:

```cs
  var contexts = _activeContextManager.ActiveContexts;
  contexts.Clear();
```

With the code as stated here, this would clear the dictionary, right? Or does `IDictionary<T>.Values` return a copy of the `Values` collection anyhow?